Dockerfile

```diff
@@ -1,32 +1,62 @@
 #
 # Dockerfile for Storm Mesos framework
 #
-FROM mesosphere/mesos:0.27.0-0.2.190.ubuntu1404
-MAINTAINER Timothy Chen <tnachen@gmail.com>
-
-ENV DEBIAN_FRONTEND noninteractive
-
-# export environment
-ENV JAVA_HOME /usr/lib/jvm/java-7-openjdk-amd64
-
-ENV MESOS_NATIVE_JAVA_LIBRARY /usr/lib/libmesos.so
-
-ARG MESOS_RELEASE=0.27.0
-ARG STORM_RELEASE=0.9.6
-ARG MIRROR=http://www.gtlib.gatech.edu/pub
+FROM ubuntu:14.04
+MAINTAINER Salimane Adjao Moustapha <me@salimane.com>
+
+# Add mesosphere, openjdk 8 repo
+# install jdk 7 and 8
+RUN DISTRO=$(lsb_release -is | tr '[:upper:]' '[:lower:]') && \
+	CODENAME=$(lsb_release -cs) && \
+	echo "deb http://repos.mesosphere.com/${DISTRO} ${CODENAME} main" > /etc/apt/sources.list.d/mesosphere.list && \
+	apt-key adv --keyserver keyserver.ubuntu.com --recv E56151BF && \
+	apt-get -y install software-properties-common && \
+	add-apt-repository ppa:openjdk-r/ppa && \
+	apt-get -y update && \
+	apt-get -y install openjdk-7-jdk openjdk-8-jdk && \
+	apt-get -yf autoremove && \
+	apt-get clean
+
+# set JAVA_HOME depending if JAVA_PRODUCT_VERSION is set
+ARG JAVA_PRODUCT_VERSION=''
+ENV TARGET_JAVA_HOME ${JAVA_PRODUCT_VERSION:+/usr/lib/jvm/java-$JAVA_PRODUCT_VERSION-openjdk-amd64}
+ENV JAVA_HOME ${TARGET_JAVA_HOME:-/usr/lib/jvm/java-7-openjdk-amd64}
+ENV JAVA_PRODUCT_VERSION ${JAVA_PRODUCT_VERSION:-7}
+
+# install mesos version, build arg, MESOS_RELEASE with a default value
+ARG MESOS_RELEASE=0.28.0
+RUN update-alternatives --install /usr/bin/java java ${JAVA_HOME%*/}/bin/java 20000 && \
```

Comment:
Can you please help me to understand precisely what this `${JAVA_HOME%*/}` syntax is doing?  I believe you are trying to remove a trailing /.  But I cannot determine what the `*` character is being used for.